app/serializers/dist/json_serializer.rb

```diff
@@ -70,5 +72,61 @@ def serialize_nested(connection)
         name: connection.name,
       }
     end
+
+    def build_mapping_blocks(mapping_rules)
+      all_mapping_blocks = []
+      puts "Generating mappings ..."
+      mapping_count = 0
+      Integration.all.each do |integration|
+        [true, false].each do |from_shopify|
+          mapping_rule_block = mapping_rules.where(
+            integration_id: integration.id,
+            from_shopify: from_shopify,
+          )
+          next if mapping_rule_block.count.zero?
+
+          mappings = []
+          Category.verticals.each do |vertical|
+            mappings << MappingBuilder.build_one_to_one_mappings_for_vertical(
+              mapping_rules: mapping_rule_block,
+              vertical: vertical,
+            )
+          end
+          processed_mappings = mappings.flatten.compact
+          mapping_count += processed_mappings.count
+          all_mapping_blocks << {
+            input_taxonomy: "shopify/v1",
+            output_taxonomy: "#{integration.name}/v1",
+            rules: processed_mappings,
+          }
+        end
+      end
```

Comment:
I had done this piece by piece, but assembling it:

```suggestion
      mapping_rules = Integration.all.flat_map do |integration|
        [true, false].filter_map do |from_shopify|
          rules = mapping_rules.where(integration:, from_shopify:)
          rules if rule.any?
        end
      end
      
      rules = Category.verticals.flat_map do |vertical|
        MappingBuilder.build_one_to_one_mappings_for_vertical(mapping_rules:, vertical:)
      end
      
      mapping_rules.map do |mapping|
        {
          input_taxonomy: "shopify/v1", # isn't this determined by the true/false technically?
          output_taxonomy: "#{mapping.integration.name}/v1",
          rules:,
        } 
      end
```

Reply:
Thanks! Simplified the code based on the suggestion.